app/views/locations/_locations.html.haml

```diff
@@ -3,12 +3,25 @@
     clearMarkers()
     clearInfoWindows();
 
+    var linky = document.getElementById("search_link");
+    var hrefOrig = linky.href;
+    var def_value = window.location.href;
+
+    if (typeof(search_string) == "undefined") {
+      search_string = def_value;
+      document.getElementById("search_link").href=search_string;
+    } else {
+      document.getElementById("search_link").href=hrefOrig + search_string;
```

Comment:
This .href line replaces the entire #search_link href. So, I first grab the existing href (pinballmap.com/portland) and put it in variable linky. Then I add it back in along with the search string.